### ISSUE TITLE
Maybe false positive from TS2801 (Promise<...> appears to always be defined)

### ISSUE THREAD

**Issue by plantain-00** [microsoft/TypeScript#45608]
# Bug Report

### 🔎 Search Terms

TS2801

### 🕗 Version & Regression Information

- This changed between versions __4.4.2__ and ___4.3.5___

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAhgLhgbxgbQNYFMCeToBOAlmAOYC6SACgSALZERYA8hJpAfDAL4wC8KbgCgiAMxgAKOGgDkcGeQCUKITBhDuQA)

### 💻 Code

```ts
const a: { [key: string]: Promise<string> } = {}
if (a['a']) {
  
}
```

### 🙁 Actual behavior

This condition will always return true since this 'Promise<string>' is always defined.

### 🙂 Expected behavior

No error.

**Comment by MartinJohns** [comment]
You want to enable `noUncheckedIndexedAccess`. Without this compiler flag it's assumed that indexed access returns a valid value.

**Comment by Alexsey** [comment]
Unfortunately, due to https://github.com/microsoft/TypeScript/issues/10530 `noUncheckedIndexedAccess` is not handy here:
```ts
if (a['a']) {
  a['a'] // Promise<string> | undefined
}
```
And I assume it will be even harder to narrow for index value `'a'` to be variable and not literal

So, the workaround here is to use `in` instead of index access:
```ts
if ('a' in a) {
  a['a'] // Promise<string>
}
```
